### PR TITLE
aws-vault 6.3.1 (new formula)

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by tresni** [comment]
Just a note that this formula currently exists in homebrew-cask, apparently due to request from the upstream author per https://github.com/Homebrew/homebrew-core/pull/3385#issuecomment-236302100

**Comment by danielnachun** [comment]
Ah, I had to do some digging to find out why, but maybe the binaries on macOS need to be code-signed by upstream rather than us?  Or was it becuase we didn't support code-signing at all in CI back in 2016?  If it's the former issue, then I guess we'd have to make this `depends_on :linux`.  Code-signing does not exist on Linux so it's not a concern there.

**Comment by danielnachun** [comment]
If you have valid AWS credentials to try this with, that would be great.  We don't have a way to test the full functionality of this tool because it requires credentials, so any help determine the code-signing situation would be great.

**Comment by tresni** [comment]
I do, I use aws-vault daily which is why I was surprised to see the PR for it here marked as “new formula”. I’ll test tomorrow morning and let you know how it goes.

**Comment by tresni** [comment]
I have no problems running aws-vault built using this formula.  I do get an oddity in that `aws-vault --version` doesn't print a string.  Looks like the build somehow sets an empty versions string.  I think that's expected based on this line `go build -ldflags="-X main.Version=" .`

May want to patch the Makefile so that we don't attempt to codesign with a non-existent certificate (especially if we codesign using a homebrew specific key later.)

```sh
homebrew/homebrew-core ●» brew reinstall --build-from-source aws-vault -v                                                                                                                                                               « [aws-vault]
Warning: Treating aws-vault as a formula. For the cask, use homebrew/cask/aws-vault
rm /usr/local/bin/aws-vault
rm /usr/local/etc/bash_completion.d/aws-vault.bash
rm /usr/local/share/zsh/site-functions/aws-vault.zsh
==> Downloading https://github.com/99designs/aws-vault/archive/v6.3.1.tar.gz
Already downloaded: /Users/bhartvig/Library/Caches/Homebrew/downloads/3da4f687fb1c22cb6a0f3011f3ffcc123043c7284329157e9a69e0bb0e178ab4--aws-vault-6.3.1.tar.gz
==> Verifying checksum for '3da4f687fb1c22cb6a0f3011f3ffcc123043c7284329157e9a69e0bb0e178ab4--aws-vault-6.3.1.tar.gz'
==> Reinstalling aws-vault
/usr/bin/sandbox-exec -f /private/tmp/homebrew20210813-94049-8gxhwr.sb nice ruby -W1 -- /usr/local/Homebrew/Library/Homebrew/build.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/aws-vault.rb --verbose
tar --extract --no-same-owner --file /Users/bhartvig/Library/Caches/Homebrew/downloads/3da4f687fb1c22cb6a0f3011f3ffcc123043c7284329157e9a69e0bb0e178ab4--aws-vault-6.3.1.tar.gz --directory /private/tmp/d20210813-94050-attjk9
cp -pR /private/tmp/d20210813-94050-attjk9/aws-vault-6.3.1/. /private/tmp/aws-vault-20210813-94050-1bq8eq/aws-vault-6.3.1
chmod -Rf +w /private/tmp/d20210813-94050-attjk9
==> make aws-vault-darwin-amd64
fatal: not a git repository (or any of the parent directories): .git
GOOS=darwin GOARCH=amd64 CGO_ENABLED=1 SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk go build -ldflags="-X main.Version=" -trimpath -o aws-vault-darwin-amd64 .
==> make install INSTALL_DIR=/usr/local/Cellar/aws-vault/6.3.1/bin
fatal: not a git repository (or any of the parent directories): .git
go build -ldflags="-X main.Version=" .
mkdir -p /usr/local/Cellar/aws-vault/6.3.1/bin
rm -f /usr/local/Cellar/aws-vault/6.3.1/bin/aws-vault
cp -a ./aws-vault /usr/local/Cellar/aws-vault/6.3.1/bin
codesign --options runtime --timestamp --sign "Developer ID Application: 99designs Inc (NRM9HVJ62Z)" /usr/local/Cellar/aws-vault/6.3.1/bin/aws-vault || true
error: The specified item could not be found in the keychain.
==> Cleaning
==> Finishing up
ln -s ../../Cellar/aws-vault/6.3.1/etc/bash_completion.d/aws-vault.bash aws-vault.bash
ln -s ../Cellar/aws-vault/6.3.1/bin/aws-vault aws-vault
ln -s ../../../Cellar/aws-vault/6.3.1/share/zsh/site-functions/aws-vault.zsh aws-vault.zsh
/usr/bin/sandbox-exec -f /private/tmp/homebrew20210813-94252-13l8z1b.sb nice ruby -W1 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/postinstall.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/aws-vault.rb
==> Caveats
zsh functions have been installed to:
  /usr/local/share/zsh/site-functions
==> Summary
🍺  /usr/local/Cellar/aws-vault/6.3.1: 7 files, 14.8MB, built in 12 seconds
homebrew/homebrew-core ●»                                                                                                                                                                                                               « [aws-vault]
``

File: Formula/aws-vault.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  depends_on "go" => :build
+
+  def install
+    os = "darwin"
+    on_linux { os = "linux" }
+    arch = Hardware::CPU.arm? ? "arm64" : "amd64"
+
+    system "make", "aws-vault-#{os}-#{arch}"
+    system "make", "install", "INSTALL_DIR=#{bin}"
```
Comment (tresni) [inline Formula/aws-vault.rb:16]:
```suggestion
    system "make", "aws-vault-#{os}-#{arch}", "VERSION=#{version}-homebrew"
    system "make", "install", "INSTALL_DIR=#{bin}", "VERSION=#{version}-homebrew"
```

This gets rid of the "Not a git repository" errors, properly sets the version, and includes something in the version string to differentiate homebrew builds from upstream.

**Comment by danielnachun** [comment]
>May want to patch the Makefile so that we don't attempt to codesign with a non-existent certificate (especially if we codesign using a homebrew specific key later.)

I'm not sure how we handle codesigning of binaries.  It's possible we need to replace the upstream certificate with our own in the Makefile.

@Homebrew/maintainers can anyone advise if we should be replacing this line with  our own certificate in the Makefile:
```
CERT_ID ?= Developer ID Application: 99designs Inc (NRM9HVJ62Z)
```
The Makefile currently signs the binary:
```
codesign --options runtime --timestamp --sign "$(CERT_ID)" $(INSTALL_DIR)/aws-vault || true
```
This does not appear to be signed on my Intel Mac at least:
```
codesign -dv --verbose=4 /usr/local/opt/aws-vault/bin/aws-vault
/usr/local/opt/aws-vault/bin/aws-vault: code object is not signed at all
```

**Comment by SMillerDev** [comment]
Brew doesn't sign binaries. So you'd just have to disable that part.

File: Formula/aws-vault.rb (hunk 1 above)
Comment (carlocab) [reply Formula/aws-vault.rb:16]: Let's do `VERSION=#{version}-#{tap.user}` in case users extract this formula to their own taps.

**Comment by danielnachun** [comment]
I've added the value for `VERSION`, and disabled the code-signing.  This should be ready for review.

**Review by SMillerDev — APPROVED** [review]

**Comment by mtibben** [comment]
I'm sure this was well-intentioned, however the `aws-vault` binary must be signed and notorized to avoid constant keychain prompts. Failing to sign the binary leads to a very poor experience when using aws-vault with macOS Keychain. I've opened a PR at https://github.com/Homebrew/homebrew-core/pull/84589 to remove this formula

**Pull request closed without merging**